File: rust_study/rust_luogu/rust_codeforces/cpp/luogu/old/P1781.cpp
/*@author    Allergy
 *@email     Allergy527@gmail.com
 *@workspace luogu\P1781.cpp
 *@data      2024/01/17 20:48:13
 */
#include<bits/stdc++.h>
#define ll long long

using namespace std;

void solve() {
    int n, ans;
    cin >> n;
    string qwq = "", x;
    for(int i = 1;i <= n;++i) {
        cin >> x;
        if(x.size() > qwq.size() || (x.size() == qwq.size()) && x > qwq) {
            qwq = x;
            ans = i;
        }
    }
    cout << ans << '\n' << qwq;
}

int main() {
    ios::sync_with_stdio(false);
    int t = 1;
    //cin >> t;
    while(t--)solve();
    cin >> t;
    return 0;
}

File: rust_luogu/src/bin/p1781.rs
//@author    Allergy
//@email     Allergy527@gmail.com
//@workspace luogu/p1781.rs
//@data      2024/01/18 12:33:31
fn cin() -> String {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).unwrap();
    input.trim().to_string()
}
fn main() {
    let t = 1;
    //1let mut input = String::new();std::io::stdin().read_line(&mut input).unwrap();let t:i32=input.parse().unwrap();
    (0..t).for_each(|_| solve());
}
fn solve() {
    let n: usize = cin().parse().unwrap();
    let comp =
        |a: &String, b: &String| -> bool { a.len() > b.len() || a.len() >= b.len() && a > b };
    let (ans, ans_idx) = (1..=n).fold((String::new(), 0), |x, i| {
        let p = cin();
        if comp(&p, &x.0) {
            (p, i)
        } else {
            x
        }
    });
    println!("{}\n{}", ans_idx, ans)
}
